lib/grp.gi

```diff
@@ -176,9 +176,13 @@ InstallMethod( IsPGroup,
         if ord = infinity then
           return false;
         elif 1 < ord then
-          UniteSet( s, Factors( ord ) );
-          if 1 < Length( s ) then
+          if not IsPrimePowerInt( s ) then
```

Comment:
Don't you mean `ord` here instead of `s` (which is a list)?

Did you check your changes at all? I would recommend adding tests cass (in the form of a new .tst file, e.g. `tst/testinstall/pgroups.tst`, which exercises this code path. (You can temporarily insert a `Print` statement here to make sure you triggered the code path).